content/docs/context.md

```diff
@@ -4,35 +4,71 @@ title: Context
->
->We provide [a codemod script](/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.proptypes) to automate the conversion.
+Context provides a way to pass data through the component tree without having to pass props down manually at every level.
+
+## Using Context
+
+Here is an example illustrating how you might inject a "theme" using context:
+
+`embed:context/theme-example.js`
+
+### API
+
+#### `React.createContext`
+
+```js
+const {Provider, Consumer} = React.createContext([default]);
+```
+
+Creates a `{ Provider, Consumer }` pair.
+
+Takes one argument, the default context that Consumers will receive when they don't have a matching Provider.
+
+
+#### `Provider`
+
+```js
+<Provider value={/* some value */}>
+```
 
-With React, it's easy to track the flow of data through your React components. When you look at a component, you can see which props are being passed, which makes your apps easy to reason about.
+A React component that allows Consumers to subscribe to context changes.
 
-In some cases, you want to pass data through the component tree without having to pass the props down manually at every level.
-You can do this directly in React with the powerful "context" API.
+Takes one prop, `value`, which will be passed to the [render prop](/docs/render-props.html) of child Consumers for the matching context anywhere in the component tree. One Provider can be connected to many Consumers.
+
+#### `Consumer`
+
+```js
+<Consumer>
+ { value => { /* render something based on the context value */ } }
+</Consumer>
+```
+
+A React component that subscribes to context changes. 
+
+Takes a function as the `children` prop that receives the `value` prop of the matching Provider. This function will be called whenever the Provider's value is updated.
 
 > Note:
 >
-> A [new, safe version of context](https://github.com/reactjs/rfcs/blob/master/text/0002-new-version-of-context.md) is under development for the upcoming 16.3 release.
+> For more information about this pattern, see [render props](/docs/render-props.html).
+
+### Typical Usage
 
+`embed:context/theme-detailed-theme-context.js`
 
-## Why Not To Use Context
+`embed:context/theme-detailed-themed-button.js`
 
-The vast majority of applications do not need to use context.
+`embed:context/theme-detailed-app.js`
 
-If you want your application to be stable, don't use context. It is an experimental API and it is likely to break in future releases of React.
 
-If you aren't familiar with state management libraries like [Redux](https://github.com/reactjs/redux) or [MobX](https://github.com/mobxjs/mobx), don't use context. For many practical applications, these libraries and their React bindings are a good choice for managing state that is relevant to many components. It is far more likely that Redux is the right solution to your problem than that context is the right solution.
+## Experimental API (Deprecated in React 16.3)
```

Comment:
Maybe this could be broken out into its own "Legacy context" page that is linked to from this page- but isn't on the side bar anywhere?